apps/prairielearn/src/lib/groups.ts

```diff
@@ -227,10 +230,21 @@ export async function addUserToGroup({
         course_instance_id: group.course_instance_id,
         user_id: user.user_id,
       }));
+
     // To be part of a group, the user needs to either be enrolled in the course
     // instance, or be an instructor
     if (!userIsStudent && !userIsInstructor) {
-      throw new GroupOperationError(`User ${uid} is not enrolled in this course.`);
+      // In the example course, any user with instructor access in any other
+      // course should have access and thus be allowed to be added to a group.
+      const course_instance = await selectCourseInstanceById(group.course_instance_id);
+      const course = course_instance ? await selectCourseById(course_instance.course_id) : null;
```

Comment:
Total tangent, but I wish we'd made two versions of all these functions: `selectCourseById()` that cannot return `null`, and `selectOptionalCourseById()` that returns `Course | null`. In code like above, it's a hard error if we can't find the user/course_instance/course and the code would be a lot cleaner without all the checks to make sure they aren't null.